src/components/molecules/collabsible-tree/index.tsx

```diff
@@ -0,0 +1,166 @@
+import React from "react"
+import clsx from "clsx"
+
+import MoreHorizontalIcon from "../../fundamentals/icons/more-horizontal-icon"
+import MinusIcon from "../../fundamentals/icons/minus-icon"
+import PlusIcon from "../../fundamentals/icons/plus-icon"
+import Button from "../../fundamentals/button"
+import Actionables, { ActionType } from "../actionables"
+
+type CollabsibleTreeProps = {
+  className?: string
+  children: React.ReactNode[]
+}
+
+type CollabsibleTreeParentProps = {
+  actions?: ActionType[]
+  open?: boolean
+  setOpen?: (v: boolean) => void
+  className?: string
+  children: React.ReactNode[]
+  __TYPE?: string
+}
+
+type CollabsibleTreeLeafProps = {
+  actions?: ActionType[]
+  className?: string
+  children: React.ReactNode[]
+  __TYPE?: string
+}
+
+
+type CollabsibleTreeType = React.FC<CollabsibleTreeProps> & {
+  Parent: React.FC<CollabsibleTreeParentProps>
+  Leaf: React.FC<CollabsibleTreeLeafProps>
+}
+
+export const CollabsibleTree: CollabsibleTreeType = ({
+  children,
+  className
+}) => {
+  const [open, setOpen] = React.useState(false)
+
+  const trigger = React.Children.toArray(children).find((child) => {
+    if (typeof child === "object" && "props" in child) {
+      return child.props.__TYPE === "CollabsibleTreeParent"
+    }
+    return false
+  })
+
+  const leaves = React.Children.toArray(children).filter((child) => {
+    if (typeof child === "object" && "props" in child) {
+      return child.props.__TYPE === "CollabsibleTreeLeaf"
+    }
+    return false
+  })
```

Comment:
I think this can be solved in a slightly better way: instead of looking for the trigger + leaves, we can make the separation explicit through the API of the component, then use it in combination with a context to control state. Something like this (removing types for conciseness):
```jsx
const CollapsibleTreeContext = React.createContext(null)

export const CollapsibleTree = ({ children }) => {
  const [open, setOpen] = React.useState(false)

  return (
    <CollapsibleTreeContext.Provider
      value={{
        open,
        handleOpen: () => setOpen(true),
        handleClose: () => setOpen(false),
        toggle: () => setOpen((open) => !open),
      }}
    >
      {children}
    </CollapsibleTreeContext.Provider>
  )
}

const useCollapsibleTree = () => {
  const context = React.useContext(CollapsibleTreeContext)
  if (!context) {
    throw new Error("useCollapsibleTree must be a child of CollapsibleTree")
  }
  return context
}

/* ---------------------------- CollapsibleTreeContent ------------------------------------ */

const CollapsibleTreeContent = ({
  children,
  className,
  ...props
}) => {
  const { open } = useCollapsibleTree()
  return (
    <div
      className={clsx(
        "pl-5 mt-xsmall",
        {
          hidden: !open,
          "animate-fade-in-top": open,
        },
        className
      )}
      {...props}
    >
      {children}
    </div>
  )
}

CollapsibleTree.Content = CollapsibleTreeContent

/* ---------------------------- CollapsibleTreeParent ------------------------------------ */

const CollapsibleTreeParent = ({
  actions,
  className,
  children,
}) => {
  const { open, toggle } = useCollapsibleTree()
  return (
    ...
  )
}

CollapsibleTree.Parent = CollapsibleTreeParent

```


Then in usage:

```jsx
const ProductVariantTree = ({ product }) => {
  return (
    <CollapsibleTree>
      <CollapsibleTree.Parent
        actions={actions}
      >
        {/* Ludvig's great designs */}
      </CollapsibleTree.Parent>
      <CollapsibleTree.Content>
        {product.variants.map((variant) => (
          <CollapsibleTree.Leaf
            actions={actions}
          >
            <ProductVariantLeaf key={variant.id} {...variant} />
          </CollapsibleTree.Leaf>
        ))}
      </CollapsibleTree.Content>
    </CollapsibleTree>
  )
}

```

Would love to hear what you think about this approach @srindom!